lib/logstash/plugin_mixins/http_client.rb

```diff
@@ -7,203 +7,230 @@
-        :truststore_password => @truststore_password.value
-      )
-      
-      if c[:ssl][:truststore_password].nil?
-        raise LogStash::ConfigurationError, "Truststore declared without a password! This is not valid, please set the 'truststore_password' option"
+    private
+
+    def ssl_options
+
+      options = {}
+      if @ssl_certificate_authorities&.any?
+        raise LogStash::ConfigurationError, 'Multiple values on `ssl_certificate_authorities` are not supported by this plugin' if @ssl_certificate_authorities.size > 1
+
+        options[:ca_file] = @ssl_certificate_authorities.first
       end
-    end
 
-    if @keystore
-      c[:ssl].merge!(
-        :keystore => @keystore,
-        :keystore_type => @keystore_type,
-        :keystore_password => @keystore_password.value
-      )
+      if @ssl_truststore_path
+        options[:truststore] = @ssl_truststore_path
+        options[:truststore_type] = @ssl_truststore_type if @ssl_truststore_type
+        options[:truststore_password] = @ssl_truststore_password.value if @ssl_truststore_password
+      elsif @ssl_truststore_password
+        fail LogStash::ConfigurationError, "truststore_password requires ssl_truststore_path you fool"
+      end
 
-      if c[:ssl][:keystore_password].nil?
-        raise LogStash::ConfigurationError, "Keystore declared without a password! This is not valid, please set the 'keystore_password' option"
+      if @ssl_keystore_path
+        options[:keystore] = @ssl_keystore_path
+        options[:keystore_type] = @ssl_keystore_type if @ssl_keystore_type
+        options[:keystore_password] = @ssl_keystore_password.value if @ssl_keystore_password
+      elsif @ssl_keystore_password
+        fail LogStash::ConfigurationError, "ssl_keystore_password requires ssl_keystore_path you fool"
       end
-    end
 
-    if @client_cert && @client_key
-      c[:ssl][:client_cert] = @client_cert
-      c[:ssl][:client_key] = @client_key
-    elsif !!@client_cert ^ !!@client_key
-      raise InvalidHTTPConfigError, "You must specify both client_cert and client_key for an HTTP client, or neither!"
-    end
+      if @ssl_certificate && @ssl_key
+        options[:client_cert] = @ssl_certificate
+        options[:client_key] = @ssl_key
+      elsif !!@ssl_certificate ^ !!@ssl_key
+        raise InvalidHTTPConfigError, "You must specify both `ssl_certificate` and `ssl_key` for an HTTP client, or neither!"
+      end
 
-    case @ssl_verification_mode
-    when 'full'
-      # NOTE: would make sense to have :browser here but historically we've used the (:strict) default
-      #
-      # supporting `ssl_verification_mode => strict` the same way ES does might require upstream Manticore
-      # changes -> as in ES/Beats setting `strict` means: "if the SAN is empty return an error"
-      c[:ssl][:verify] = :strict # :default
-    when 'none'
-      c[:ssl][:verify] = :disable
-    end
+      options[:verify] = @ssl_verification_mode == 'full' ? :strict : :disable
```

Comment:
Likely should file a separate issue, but `:strict` is deprecated and the poorly-named `:default` is the modern RFC-compliant strict verifier (see: https://github.com/logstash-plugins/logstash-output-elasticsearch/pull/1139)